codegen/generate.go

```
@@ -294,17 +294,61 @@ func RemoteStateConfigToTerraformCode(backend string, config map[string]any, enc
 			}
-			// extract assume role hcl values to be rendered in HCL
-			var assumeRoleMap map[string]string
+			// Extracting the values requires two steps.
+			// Parsing into a struct first, enabling hclsimple.Decode() to deal with complex types.
+			// Then copying values into the assumeRoleMap for rendering to HCL.
+			var assumeRoleMap map[string]any
+			assumeRoleMap = make(map[string]any)
+			type assumeRoleConfig struct {
+				RoleArn           string            `hcl:"role_arn"`
+				Duration          string            `hcl:"duration,optional"`
+				ExternalId        string            `hcl:"external_id,optional"`
```

golangci-lint (1.62.2)

ST1003: struct field ExternalId should be ExternalID

(stylecheck)